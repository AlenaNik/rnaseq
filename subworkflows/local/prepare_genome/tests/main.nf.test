nextflow_workflow {

    name "Test Subworkflow PREPARE_GENOME"
    script "../main.nf"
    workflow "PREPARE_GENOME"
    config "./nextflow.config"

    tag "PREPARE_GENOME"

    tag "BBMAP_BBSPLIT"
    tag "CUSTOM_CATADDITIONALFASTA"
    tag "CUSTOM_GETCHROMSIZES"
    tag "GFFREAD"
    tag "GTF2BED"
    tag "GTF_FILTER"
    tag "GUNZIP"
    tag "HISAT2_BUILD"
    tag "HISAT2_EXTRACTSPLICESITES"
    tag "KALLISTO_INDEX"
    tag "PREPROCESS_TRANSCRIPTS_FASTA_GENCODE"
    tag "RSEM_PREPAREREFERENCE"
    tag "RSEM_PREPAREREFERENCE"
    tag "SALMON_INDEX"
    tag "SORTMERNA"
    tag "STAR_GENOMEGENERATE"
    tag "STAR_GENOMEGENERATE_IGENOMES"
    tag "UNTAR"

    test("star - with igenomes") {

        when {
            workflow {
                """
                gencode                  = false
                featurecounts_group_type = 'gene_biotype'
                aligner                  = 'star_salmon,star_rsem,hisat2'
                pseudo_aligner           = 'salmon'
                skip_gtf_filter          = true
                skip_bbsplit             = true
                skip_sortmerna           = true
                skip_alignment           = true
                skip_pseudo_alignment    = true

                input[0] = file(params.pipelines_testdata_base_path + 'reference/genome.fasta', checkIfExists: true)
                input[1] = file(params.pipelines_testdata_base_path + 'reference/genes_with_empty_tid.gtf', checkIfExists: true)
                input[2] = file(params.pipelines_testdata_base_path + 'reference/genes.gff', checkIfExists: true)
                input[3] = file(params.pipelines_testdata_base_path + 'reference/gfp.fa', checkIfExists: true)
                input[4] = file(params.pipelines_testdata_base_path + 'reference/transcriptome.fasta', checkIfExists: true)
                input[5] = null
                input[6] = null
                input[7] = file(params.pipelines_testdata_base_path + 'reference/bbsplit_fasta_list.txt', checkIfExists: true)
                input[8] = null
                input[9] = null
                input[10] = file(params.pipelines_testdata_base_path + 'reference/rsem.tar.gz', checkIfExists: true)
                input[11] = file(params.pipelines_testdata_base_path + 'reference/salmon.tar.gz', checkIfExists: true)
                input[12] = null
                input[13] = file(params.pipelines_testdata_base_path + 'reference/hisat2.tar.gz', checkIfExists: true)
                input[14] = null
                input[15] = null
                input[16] = gencode
                input[17] = featurecounts_group_type
                input[18] = aligner
                input[19] = pseudo_aligner
                input[20] = skip_gtf_filter
                input[21] = skip_bbsplit
                input[22] = skip_sortmerna
                input[23] = skip_alignment
                input[24] = skip_pseudo_alignment
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(file(workflow.out.transcript_fasta[0][1]).name).match("transcript_fasta") },
                { assert snapshot(file(workflow.out.salmon_index[0][1]).name).match("salmon_index") },
                { assert snapshot(workflow.out.fasta).match("fasta") },
                { assert snapshot(workflow.out.gtf).match("gtf") },
                { assert snapshot(workflow.out.fai).match("fai") },
                { assert snapshot(workflow.out.gene_bed).match("gene_bed") },
                { assert snapshot(workflow.out.chrom_sizes).match("chrom_sizes") },
                { assert snapshot(workflow.out.splicesites).match("splicesites") },
                { assert snapshot(workflow.out.bbsplit_index).match("bbsplit_index") },
                { assert snapshot(workflow.out.sortmerna_index).match("sortmerna_index") },
                { assert snapshot(workflow.out.star_index).match("star_index") },
                { assert snapshot(workflow.out.rsem_index).match("rsem_index") },
                { assert snapshot(workflow.out.hisat2_index).match("hisat2_index") },
                { assert snapshot(workflow.out.kallisto_index).match("kallisto_index") },
                { assert snapshot(workflow.out.versions).match("versions") }
            )
        }
    }
}
