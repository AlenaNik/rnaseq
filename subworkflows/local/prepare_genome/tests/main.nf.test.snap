{
    "bbsplit_index": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T15:59:35.434161"
    },
    "gene_bed": {
        "content": [
            [
                "genome_gfp.bed:md5,991993ebef1def1d5823632c21177ec3"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T15:59:35.394732"
    },
    "fasta": {
        "content": [
            [
                "genome_gfp.fasta:md5,e23e302af63736a199985a169fdac055"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T15:59:35.356464"
    },
    "fai": {
        "content": [
            [
                "genome_gfp.fasta.fai:md5,8fa54c6bd2ea6a369efbb8ab4f30156a"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T15:59:35.376112"
    },
    "splicesites": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T15:59:35.424874"
    },
    "chrom_sizes": {
        "content": [
            [
                "genome_gfp.fasta.sizes:md5,9b755f8f349b14accefb4d859f84de26"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T15:59:35.412903"
    },
    "star_index": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T15:59:35.452927"
    },
    "versions": {
        "content": [
            [
                "versions.yml:md5,1cd265cb86a23a27f683ffb2459f37f5",
                "versions.yml:md5,58fa668eeacb984c10df6f856f2c9291",
                "versions.yml:md5,71252f1a221be05593361acccb99506b"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T15:59:35.490956"
    },
    "gtf": {
        "content": [
            [
                "genome_gfp.gtf:md5,5e63cda3dcafc0d6ad0a738133af9d54"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T15:59:35.363682"
    },
    "kallisto_index": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T15:59:35.481666"
    },
    "sortmerna_index": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T15:59:35.44388"
    },
    "transcript_fasta": {
        "content": [
            "n"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T15:59:35.35063"
    },
    "salmon_index": {
        "content": [
            "n"
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T15:59:35.353967"
    },
    "rsem_index": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T15:59:35.462738"
    },
    "hisat2_index": {
        "content": [
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-03-08T15:59:35.472712"
    }
}